### PR TITLE
Multiple scoring support of `OptunaSearchCV`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

https://github.com/optuna/optuna/issues/3521. This feature was requested a few years ago https://github.com/optuna/optuna/issues/2935

## Description of the changes

For discussion, let me share the current approach to resolving the issue even though mypy is unhappy with my code changes. I suppose this module is not well-maintained due to maintenance costs, so I'm still not sure introducing more complicated features like this PR is beneficial for Optuna.. Any comments are welcome!

### CODE REVIEW

File: optuna/integration/sklearn.py [hunk 1]
```diff
@@ -32,21 +33,17 @@
...
     from sklearn.model_selection import check_cv
     from sklearn.model_selection import cross_validate
+    from sklearn.utils import _safe_indexing as sklearn_safe_indexing
     from sklearn.utils import check_random_state
     from sklearn.utils.metaestimators import _safe_split
-
-    if sklearn.__version__ >= "0.22":
-        from sklearn.utils import _safe_indexing as sklearn_safe_indexing
-    else:
-        from sklearn.utils import safe_indexing as sklearn_safe_indexing
```
Comment (nzw0301) [inline optuna/integration/sklearn.py:49]: https://github.com/optuna/optuna/blob/82324f190a69d86058e24082fde6a2f6a6da3a1a/setup.py#L103

**Comment by nzw0301** [comment]
Okay, no one is interested in this PR; let me close this.

**Comment by HideakiImamura** [comment]
I'm sorry that we didn't notice your PR earlier. I apologize if this has caused any inconvenience. I'll reopen the PR, and if possible, could you continue your work on it? If you're short on time, we can take over this task on our end.

**Comment by nzw0301** [comment]
No worries. Unfortunately, I completely forgot this PR's content so please feel free to close this PR or take over this PR if this feature is necessary.

**Comment by HideakiImamura** [comment]
Thank you for your response. So, we will close this PR and mark the task itself as contribution-welcome.

**Pull request closed without merging**